daos/message-dao.ts

```diff
@@ -24,65 +25,119 @@ export default class MessageDao implements MessageDaoI {
         }
         return MessageDao.messageDao;
     }
-
     private constructor() {}
 
     /**
-     * Uses MessageModel to retrieve all message documents received by the user
+     * Uses MessageModel to retrieve all message documents
+     * @returns Promise To be notified when the messages are retrieved from database
+     */
+    findAllMessages = async (): Promise<Message[]> =>
+        MessageModel.find()
+            .populate("sender")
+            .populate("recipient")
+            .exec();
+
+    /**
+     * Uses MessageModel to retrieve all message documents sent by the user
      * from message collection
      * @param {string} uid User's primary key
      * @returns Promise To be notified when the messages are retrieved from database
      */
-    findAllMessagesReceivedByUser = async (uid: string): Promise<Message[]> => {
-        return MessageModel.find({to: uid}).populate("from");
-    }
+    findAllMessagesSentByUser = async (uid: string): Promise<Message[]> =>
+        MessageModel
+            .find({sender: uid})
+            .populate("sender")
+            .populate("recipient")
+            .exec();
```

Comment:
Rather than this, let's have a function `findMessagesSentByUser(senderUid: string, receiverUid: string)` that finds the messages sent by a user to another user which we can use with the app.get("/api/users/:senderuid/messages/:receiveruid") API.

Reply:
I guess we already have that in the method findMessagesBetweenSpecificUsers. It uses the same API path.

Reply:
Oh right! I'm sorry! Then I don't think we need these functions since we can use the function `findMessagesBetweenSpecificUsers` everywhere.